addons/mass_mailing_sms/models/mailing_mailing.py

```diff
@@ -367,3 +369,36 @@ def _get_default_ab_testing_campaign_values(self, values=None):
                 campaign_values['name'] = _("A/B Test: %s", sms_subject)
             campaign_values['ab_testing_sms_winner_selection'] = self.ab_testing_sms_winner_selection
         return campaign_values
+
+    def get_chars_counts(self):
+        """Get character counts for sms widget computation of characters and sms counts.
+
+        :return: Character counts used for links, formatted as `{optOutLink: int, link: int}`.
+        """
+        # opt-out is formatted as "\nSTOP SMS: [base_url]/sms/[mailing_id]/[trace_code]"
+        # links are formatted as "[base_url]/r/[link_tracker_code]/s/[sms_id]"
+        # Note: `url_join` will add a '/' as necessary.
+        if len(self) > 1:
+            return UserError("Call on zero or one record.")
+
+        self.check_access_rights('write')
+        if self.id:
+            mailing_id = self.id
+        else:
+            mailing_id = (self.env['mailing.mailing'].sudo().search([], order="id desc", limit=1).id or 0) + 1
```

Comment:
bit of nitpicking but I am uneasy with this non-existing id, the rest of the function only uses it inside of `len()` computation which is ok but I can imagine someone `browse`ing that ID in the future, not realizing that it is not safe.
suggestion: rename the variable as `mailing_id_hint`